course3/homework-linkedlist/linkedlist.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+#include <iostream>
+#include <memory>
+
+namespace mau {
+
+// Reverse iterator. http://stackoverflow.com/questions/8542591/c11-reverse-range-based-for-loop
+template <typename T>
+struct reversion_wrapper { T& iterable; };
```

Comment:
Interesting! Good idea.
But you could've use ``` boost::adaptors::reversed``` : http://www.boost.org/doc/libs/1_58_0/libs/range/doc/html/range/reference/adaptors/reference/reversed.html